aiobafi6/device.py

```diff
@@ -434,32 +452,23 @@ def _stop(self) -> None:
         elif self._connect_task is not None:
             self._connect_task.cancel()
         # Otherwise, if `_connect` is scheduled, cancel that and call
-        # `_sched_connect_or_signal_run_fut` directly because nothing else will.
+        # `_sched_connect_or_reset` directly because nothing else will.
         elif self._connect_timer is not None:
             self._connect_timer.cancel()
             self._connect_timer = None
-            self._sched_connect_or_signal_run_fut()
-
-    def _finish_run(self, _: asyncio.Future) -> None:
-        """Reset the run future to None.
-
-        This is the only completion callback for the run future and the only place where
-        it is reset to None, indicating that the device has fully stopped and could be
-        run again."""
-        _LOGGER.debug("%s: Stopped.", self.name)
-        self._run_fut = None
-        self._stop_requested = False
+            self._sched_connect_or_reset()
 
     # Availability
 
     @property
     def available(self) -> bool:
         """Return True when device is running and has values for critical properties."""
-        return self._available_event.is_set()
++        available_fut = self._available_fut
++        return available_fut.done() and not available_fut.exception()
```

Comment:
```suggestion
        available_fut = self._available_fut
        return available_fut.done() and not available_fut.exception()
```

Reply:
Haha, no worries. I was about to fix that.

Reply:
Thats what happens when you type before coffee in the morning :)